app/src/main/java/org/schabi/newpipe/settings/SettingsActivity.java

```diff
@@ -91,14 +171,211 @@ public boolean onOptionsItemSelected(final MenuItem item) {
     @Override
     public boolean onPreferenceStartFragment(final PreferenceFragmentCompat caller,
                                              final Preference preference) {
-        final Fragment fragment = Fragment
-                .instantiate(this, preference.getFragment(), preference.getExtras());
+        showSettingsFragment(instantiateFragment(preference.getFragment()));
+        return true;
+    }
+
+    private Fragment instantiateFragment(@NonNull final String className) {
+        return getSupportFragmentManager()
+                .getFragmentFactory()
+                .instantiate(this.getClassLoader(), className);
```

Comment:
Is `preference.getExtras()` useless? Add a comment if that's the case

Reply:
The method which used the extras was deprecated, so I removed it altogether.
I also didn't see a difference when I removed it so I'm pretty sure that it was useless 😄
![grafik](https://user-images.githubusercontent.com/40789489/148685996-64a42e83-9cd4-4322-b5fe-5322422f09db.png)

Also:
https://stackoverflow.com/a/14987257
> > what is the purpose of getExtras()?
>
> It doesn't really do anything useful. Seriously.

🤣